pkg/tfgen/generate_schema_test.go

```diff
@@ -438,6 +438,17 @@ func TestNestedDescriptions(t *testing.T) {
 	bridgetesting.AssertEqualsJSONFile(t, "test_data/nested-descriptions-schema.json", schema)
 }
 
+func TestRequiredInputWithDefault(t *testing.T) {
+	t.Parallel()
+	provider := testprovider.ProviderRequiredInputWithDefaultFunc()
+	schema, err := GenerateSchema(provider, diag.DefaultSink(io.Discard, io.Discard, diag.FormatOptions{
+		Color: colors.Never,
+	}))
+	require.NoError(t, err)
+	require.Empty(t, schema.Resources["testprovider:index:Res"].RequiredInputs)
+	bridgetesting.AssertEqualsJSONFile(t, "test_data/required-input-with-default-schema.json", schema)
```

Comment:
A little more precise assert would be appreciated in addition to the generic assert, make the test case readable. You are looking at some property becoming optional right? Can you assert on that?